components/DeleteButton.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import { Button, Text } from "@mantine/core";
+import { cleanNotifications, showNotification, NotificationProps } from "@mantine/notifications";
+import { Check, X } from "tabler-icons-react";
+import { useModals } from "@mantine/modals";
+
+/**
+ * DeleteButton is a component for rendering a delete resource button
+ * @returns Button that, when clicked, displays a notification verifying that the user
+ * wants to delete the resource. Only deletes if the user confirms, cancels otherwise.
+ */
+export default function DeleteButton() {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const router = useRouter();
+  const modals = useModals();
+  const { resourceType, id } = router.query;
+
+  let customMessage: NotificationProps["message"] = <div>Problem connecting to server</div>;
+  let notifProps: NotificationProps = {
+    message: customMessage,
+    color: "red",
+    icon: <X size={18} />,
+    autoClose: false,
+  };
+
+  const deleteHandler = () => {
+    //delete the resource from the server
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}/${id}`, {
+      method: "DELETE",
+    })
+      .then((response) => {
+        if (response.status === 201 || response.status === 200 || response.status === 204) {
```

Comment:
```suggestion
        if (response.status === 204) {
```
Only status code 204 is applicable here because 200 is OK and 201 is Created where 204 is No content. You can read more here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status